expression/column.go

```diff
@@ -308,19 +309,21 @@ func (col *Column) HashCode(_ *stmtctx.StatementContext) []byte {
 }
 
 // ResolveIndices implements Expression interface.
-func (col *Column) ResolveIndices(schema *Schema) {
-	col.Index = schema.ColumnIndex(col)
+func (col *Column) ResolveIndices(schema *Schema) Expression {
+	newCol := col.Clone().(*Column)
+	newCol.Index = schema.ColumnIndex(col)
 	// If col's index equals to -1, it means a internal logic error happens.
-	if col.Index == -1 {
+	if newCol.Index == -1 {
 		log.Errorf("Can't find column %s in schema %s", col, schema)
```

Comment:
please remove this log, in stead we should return this error and handle it outside, not just logging it and let the following execution panic

Reply:
I think outside cannot handle it too.
How about just panic here?

Reply:
outside can handle it, see `physicalOptimize`